src/scala/com/github/johnynek/bazel_deps/MakeDeps.scala

```diff
@@ -92,24 +92,50 @@ object MakeDeps {
             System.exit(-1)
             sys.error("exited already")
         }
-        // Build up the IO operations that need to run. Till now,
-        // nothing was written
-        val io = for {
-          _ <- IO.recursiveRmF(IO.Path(model.getOptions.getThirdPartyDirectory.parts))
-          wsp = IO.path(workspacePath)
-          _ <- IO.mkdirs(wsp.parent)
-          _ <- IO.writeUtf8(wsp, ws)
-          _ <- IO.writeUtf8(wsp.sibling("BUILD"), "")
-          builds <- Writer.createBuildFiles(model.getOptions.getBuildHeader, targets)
-        } yield builds
 
-        // Here we actually run the whole thing
-        io.foldMap(IO.fileSystemExec(projectRoot.toFile)) match {
-          case Failure(err) =>
-            System.err.println(err)
-            System.exit(-1)
-          case Success(builds) =>
-            println(s"wrote ${targets.size} targets in $builds BUILD files")
+        if (g.checkOnly) {
+          // Build up the IO operations that need to run.
+          val wsp = IO.path(workspacePath)
```

Comment:
can we move this to a method in `object MakeDeps` which takes some arguments and returns an `IO[Unit]` I guess? Same with the else condition?

Reply:
Move which part?  The IO operations and the response handling are different in the two branches, so there's little shared code.

Reply:
yeah, the method is just getting long. Can we have methods to compartmentalize the logic at least?

Reply:
moved the two branches to their own helper methods.  they have identical, longish signatures -- is it idiomatic to make a case class to capture all those params, or something else to clean them up? or is that ok as-is?